commons/protocol/src/main/scala/sbt/protocol/Protocol.scala

```diff
@@ -217,54 +215,46 @@ final case class BackgroundJobLogEvent(jobId: Long, entry: LogEntry) extends Log
   require(jobId != 0L)
 }
 
-/** A custom event from a task. "name" is conventionally the simplified class name. */
-final case class TaskEvent(taskId: Long, name: String, serialized: SerializedValue) extends Event
+/** A custom event from a task. */
+final case class TaskEvent(taskId: Long, serialized: SerializedValue) extends Event
 
 object TaskEvent {
   def apply[T: SPickler](taskId: Long, event: T): TaskEvent = {
     val serialized = SerializedValue(event)
-    TaskEvent(taskId, MessageSerialization.makeSimpleName(event.getClass), serialized)
+    TaskEvent(taskId, serialized)
   }
 }
 
 /** Companion objects of events which can go in a task event extend this */
 trait TaskEventUnapply[T] {
-  import scala.reflect.ClassTag
-
-  def unapply(event: Event)(implicit unpickler: Unpickler[T], classTag: ClassTag[T]): Option[(Long, T)] = event match {
+  def unapply(event: Event)(implicit unpickler: Unpickler[T]): Option[(Long, T)] = event match {
     case taskEvent: TaskEvent =>
-      val name = MessageSerialization.makeSimpleName(implicitly[ClassTag[T]].runtimeClass)
-      if (name != taskEvent.name) {
+      if (taskEvent.serialized.hasTag[T])
+        Some(taskEvent.taskId -> taskEvent.serialized.parse[T].get)
```

Comment:
That was already effectively required if you used the default "name" field generated from the class name, and I think SendEventService always used that (there's no way to give SendEventService a different name). With this change I guess you could avoid that requirement by writing custom code, but not sure anyone will bother. I guess the way to make this more robust would be a method on Unpickler, like canUnpickleTag. Anyway I don't think we're making things worse than they were.